src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.cpp

```diff
@@ -0,0 +1,821 @@
+
+        real64 N[numNodesPerElem];
+        real64 gradN[ numNodesPerElem ][ 3 ];
+
+        /// Loop over elements
+        for( localIndex k=0; k < elemsToNodes.size( 0 ); ++k )
+        {
+          real64 const invC2 = 1.0 / ( c[k] * c[k] );
+          real64 xLocal[numNodesPerElem][3];
+          for( localIndex a=0; a< numNodesPerElem; ++a )
+          {
+            for( localIndex i=0; i<3; ++i )
+            {
+              xLocal[a][i] = X( elemsToNodes( k, a ), i );
+            }
+          }
+
+          for( localIndex q=0; q<numQuadraturePointsPerElem; ++q )
+          {
+            FE_TYPE::calcN( q, N );
+            real64 const detJ = finiteElement.template getGradN< FE_TYPE >( k, q, xLocal, gradN );
+
+            for( localIndex a=0; a< numNodesPerElem; ++a )
+            {
+              mass[elemsToNodes[k][a]] +=  invC2 * detJ * N[a];
+            }
+          }
+
+          real64 const alpha = 1.0/c[k];
+
+          for( localIndex kfe=0; kfe< numFacesPerElem; ++kfe )
+          {
+            localIndex const numFaceGl = elemsToFaces[k][kfe];
+
+            /// Face on the domain boundary and not on free surface
+            if( facesDomainBoundaryIndicator[numFaceGl]==1 && freeSurfaceFaceIndicator[numFaceGl]!=1 )
+            {
+              for( localIndex q=0; q<numQuadraturePointsPerElem; ++q )
+              {
+                FE_TYPE::calcN( q, N );
+                real64 const detJ = finiteElement.template getGradN< FE_TYPE >( k, q, xLocal, gradN );
+
+                real64 invJ[3][3]={{0}};
+                FE_TYPE::invJacobianTransformation( q, xLocal, invJ );
+
+                for( localIndex a=0; a < numNodesPerFace; ++a )
+                {
+                  /// compute ds=||detJ*invJ*normalFace_{kfe}||
+                  real64 tmp[3]={0};
+                  real64 ds = 0.0;
+                  for( localIndex i=0; i<3; ++i )
+                  {
+                    for( localIndex j = 0; j < 3; ++j )
+                    {
+                      tmp[i] += invJ[j][i]*faceNormal[numFaceGl][j];
+                    }
+                    ds +=tmp[i]*tmp[i];
+                  }
+                  ds = std::sqrt( ds );
+
+                  localIndex numNodeGl = facesToNodes[numFaceGl][a];
+                  damping[numNodeGl] += alpha*detJ*ds*N[a];
+                }
+              }
+            }
+          } // end loop over element
+
+          /* Unit test
+                 // Test for mass matrix sumTerm*c2 should be volume of the domaine
+                    real64 sumMass = 0.0;
+                    for( localIndex a=0; a<nodeManager.size(); ++a )
+                    {
+                      sumMass +=mass[a];
+                    }
+
+                 // assuming MediumVelocity c = 1500
+                 sumMass *=1500*1500;
+                 std::cout << "Sum mass terms time C2 = " << sumMass << std::endl;
+
+                 GEOSX_ERROR_IF( true, " Stop test Mass Ok" );
+           */
```

Comment:
I suggest to remove this as well. You validated your results against Garcimore, and when you merge you are going to save the baselines of your integrated test(s), so I don't think this is needed anymore.

Reply:
Done !